cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaSetOperator.java

```diff
@@ -73,4 +73,7 @@ public static boolean needsRollingUpdate(StatefulSetDiff diff) {
                 .rollingRestart(podNeedsRestart);
     }
 
+    public AdminClientProvider getAdminClientProvider() {
```

Comment:
This seems a bit like a dirty hack. Why not pass the AdminClientProvider through the `KafkaAssemblyOperator` and `ResourceOperatorSupplier`?